python/pypto/runtime/runner.py

```
@@ -869,3 +869,100 @@ def _add_headers_to_file(cpp_file: Path) -> None:
+    # 1. Ensure orchestration headers are patched (idempotent)
+    _patch_orchestration_headers(work_dir)
+
+    # 2. Generate minimal golden.py for CodeRunner
+    golden_path = work_dir / "_call_golden.py"
+    output_names = [param_infos[i].name for i in output_indices]
+    _write_call_golden(golden_path, param_infos, output_names)
+
+    # 3. Save user tensors to cache so the golden_inputs_patch can load them
+    cache_dir = work_dir / "cache"
+    cache_dir.mkdir(exist_ok=True)
+    inputs_data = [(info.name, tensor) for info, tensor in zip(param_infos, tensors)]
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)